### PR TITLE
#1968

### PR DESCRIPTION
**Pull request by JayasilanJibm**
Vlan Unit Test

Hi Chris,

I have written the unit test cases for vlans and got 100 % coverage. Please review and approve this pull request.

Attaching the screenshot of coverage report for further reference.
image

Title  - Vlan unit tests 100% (Issue #1968)

Linked Issue #1968

Thanks,
Jayasilan

### CODE REVIEW

File: tests/CLI/modules/vlan_tests.py [hunk 1]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+        self.assert_called_with('SoftLayer_Network_Pod', 'getAllObjects')
+        self.assertIsInstance(result.exception, ex.SoftLayerError)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_create_option(self, confirm_mock):
+        confirm_mock.return_value = True
+        datacenter_mock = self.set_mock('SoftLayer_Location_Datacenter', 'getDatacenters')
+        datacenter_mock.return_value = [{'id': 814994,
+                                         'name': 'ams03'
+                                         }]
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:152]: This could be 1 line

File: tests/CLI/modules/vlan_tests.py [hunk 2]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_list_get_pod_with_closed_announcement(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Account', 'getNetworkVlans')
+        gpods_mock = self.set_mock('SoftLayer_Network_Pod', 'getAllObjects')
+        vlan_mock.return_value = {'primaryRouter': {
+            'fullyQualifiedDomainName': 'bcr01a.fra02.softlayer.com',
+            'id': 462912},
+                                  'tagReferences': ''}
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:167]: Fix the formatting here, tagReferences should line up with id

File: tests/CLI/modules/vlan_tests.py [hunk 3]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_list_get_pod_with_closed_announcement_no_closure(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Account', 'getNetworkVlans')
+        gpods_mock = self.set_mock('SoftLayer_Network_Pod', 'getAllObjects')
+        vlan_mock.return_value = {
+            'primaryRouter': {'fullyQualifiedDomainName': 'bcr01a.fra02.softlayer.com',
+                              'id': 462912
+                              },
+            'tagReferences': ''}
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:188]: Formatting

File: tests/CLI/modules/vlan_tests.py [hunk 4]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+                                    'name': 'fra02.pod01',
+                                    'capabilities': ['SUPPORTS_SECURITY_GROUP']}]
+        result = self.run_command(['vlan', 'list'])
+        self.assert_no_fail(result)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_details_no_vs_no_hard_no_trunck(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Network_Vlan', 'getObject')
+        get_object = {'primaryRouter': {'fullyQualifiedDomainName': 'bcr02a.fra02.softlayer.com',
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:204]: For large mock object likes this, I would prefer to put them in a fixture. https://github.com/softlayer/softlayer-python/blob/master/tests/managers/block_tests.py#L433 is one example of that pattern.

Basically you would do the following:

1. `from SoftLayer.fixtures import SoftLayer_Network_Vlan`
2. in the [SoftLayer/fixtures/SoftLayer_Network_Vlan.py](https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/fixtures/SoftLayer_Network_Vlan.py) file you would create a variable like `getObject_no_trunk = <what you have for get_object here>`
3. `vlan_mock.return_value = SoftLayer_Network_Vlan.getObject_no_trunk`

That helps make these tests a little easier to read. You only need to do that for long objects like this one though. The other mocks you have in these tests are fine.

File: tests/CLI/modules/vlan_tests.py [hunk 5]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+                                    'datacenterLongName': 'Frankfurt 2',
+                                    'frontendRouterId': 335916,
+                                    'frontendRouterName': 'fcr01a.fra02',
+                                    'name': 'fra02.pod01',
+                                    'capabilities': ['SUPPORTS_SECURITY_GROUP']}]
+        result = self.run_command(['vlan', 'list'])
+        self.assert_no_fail(result)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_details_no_vs_no_hard_no_trunck(self, ngb_mock):
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:201]:
```suggestion
    def test_vlan_details_no_vs_no_hard_no_trunk(self, ngb_mock):
```

File: tests/CLI/modules/vlan_tests.py [hunk 6]
```diff
@@ -204,3 +361,16 @@ def test_vlan_cancel_fail(self, confirm_mock):
...
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_cancel_exception(self, confirm_mock):
+        confirm_mock.return_value = True
+        cancel_failure_mock = self.set_mock('SoftLayer_Network_Vlan', 'getCancelFailureReasons')
+        cancel_failure_mock.return_value = []
+        vlan_mock = self.set_mock('SoftLayer_Network_Vlan', 'getObject')
+        vlan_mock.return_value = {}
+        result = self.run_command(['vlan', 'cancel', '1234'])
+        self.assertEqual(result.exit_code, 2)
+        print(result.exception)
```
Comment (allmightyspiff) [inline tests/CLI/modules/vlan_tests.py:374]: Remove this print

File: tests/CLI/modules/vlan_tests.py [hunk 7]
```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
...
+    def test_vlan_create_option(self, confirm_mock):
+        confirm_mock.return_value = True
+        datacenter_mock = self.set_mock('SoftLayer_Location_Datacenter', 'getDatacenters')
+        datacenter_mock.return_value = [{'id': 814994,
+                                         'name': 'ams03'
+                                         }]
+        router_mock = self.set_mock('SoftLayer_Location_Datacenter', 'getHardwareRouters')
+        router_mock.return_value = [{'hostname': 'bcr01a.ams03'}]
+        result = self.run_command(['vlan', 'create-options'])
+        print(result.output)
```
[the remark this answers is not in the document]
Comment (JayasilanJibm) [reply tests/CLI/modules/vlan_tests.py]: Removed the print statement

File: tests/CLI/modules/vlan_tests.py (hunk 1 above)
Comment (JayasilanJibm) [reply tests/CLI/modules/vlan_tests.py]: Changed the mock return value to  1 liner

File: tests/CLI/modules/vlan_tests.py (hunk 2 above)
Comment (JayasilanJibm) [reply tests/CLI/modules/vlan_tests.py]: Changed the formatting.

File: tests/CLI/modules/vlan_tests.py (hunk 5 above)
Comment (JayasilanJibm) [reply tests/CLI/modules/vlan_tests.py]: Changed the function name as well

**Comment by ramkishor-ch** [comment]
Hi,

I have "tested the changes".
Please find the screenshots for reference.

Screen shot: 1
Screenshot 2023-06-12 at 1 40 10 PM

Screen shot: 2
Screenshot 2023-06-12 at 1 38 52 PM


Thanks,
Ramkishor Chaladi.

**Pull request merged**